src/Npgsql/NpgsqlConnector.cs

```diff
@@ -564,13 +566,9 @@ internal async Task Open(NpgsqlTimeout timeout, bool async, CancellationToken ca
             TypeMapper.Bind(DatabaseInfo);
         }
 
-        internal async ValueTask UpdateServerType(NpgsqlTimeout timeout, bool async,
-            CancellationToken cancellationToken = default)
+        internal async ValueTask<ClusterState> UpdateServerType(
+            NpgsqlTimeout timeout, bool async, CancellationToken cancellationToken = default)
         {
-            // Extended query protocol not supported in a replication connection
```

Comment:
That's definitely not something we should remove, as otherwise it fails for the replication connection.

Reply:
I've added validation to disallow multiple hosts for replication, just like for multiplexing - at least for now. I'm not sure of the possible consequences. IIRC we don't even support pooling with replication connections...

Reply:
Oh, the tests have helped me understand :) Brought back the check.